cmd/openstack-populator/openstack-populator.go

```diff
@@ -186,14 +186,14 @@ func writeData(reader io.ReadCloser, file *os.File, imageID string, progress *pr
 
 func getAuthType() (authType clientconfig.AuthType, err error) {
 	if configuredAuthType := getStringFromSecret(authTypeString); configuredAuthType == "" {
-		authType = clientconfig.AuthPassword
+		return clientconfig.AuthPassword, nil
```

Comment:
When returning inside a condition we usually try avoiding unnecessary nesting to improve readability. In this case we can leave the else if since it's using `configuredAuthType`, but I'd drop the else:

```go
if configuredAuthType := getStringFromSecret(authTypeString); configuredAuthType == "" {
		return clientconfig.AuthPassword, nil
} else if supportedAuthType, found := supportedAuthTypes[configuredAuthType]; found {
		return supportedAuthType, nil
}
err = errors.New(unsupportedAuthTypeErrStr)
klog.Fatal(err.Error(), "authType", configuredAuthType)
return "", err
```

Reply:
Makes sense! I was hesitant to touch more than the return type, but I'll allow myself to refactor a bit around these.